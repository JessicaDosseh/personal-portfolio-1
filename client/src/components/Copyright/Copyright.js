import React from "react";
import styled, { css } from "styled-components";

const Copyright = ({ darkmode }) => {
  return (
    <CopyrightContainer darkmode={darkmode}>
      <h4>Copyright Â© 2019 Nathan Thomas</h4>
    </CopyrightContainer>
  );
};

const CopyrightContainer = styled.div`
  max-width: 1200px;
  width: 100%;
  margin: 0 auto;
  display: flex;
  justify-content: center;

  h4 {
    display: block;
    width: 100%;
    text-align: center;
    font-size: 1.6rem;
    line-height: 1.8;
    margin-bottom: 80px;
    color: black;

    @media (min-width: 500px) {
      font-size: 1.6rem;
    }

    ${props =>
      props.darkmode &&
      css`
        color: white;
      `}
  }
`;

export default Copyright;
