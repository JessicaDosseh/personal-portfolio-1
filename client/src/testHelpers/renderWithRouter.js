import React from "react";
import { Router } from "react-router-dom";
import { render } from "@testing-library/react";
import { createMemoryHistory } from "history";
import { StateProvider } from "react-conflux";
import { globalReducer, globalContext } from "../store/reducers/globalReducer";

const renderWithRouter = (
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...render(
      <StateProvider reducer={globalReducer} stateContext={globalContext}>
        <Router history={history}>{ui}</Router>
      </StateProvider>
    ),
    history
  };
};

export default renderWithRouter;
