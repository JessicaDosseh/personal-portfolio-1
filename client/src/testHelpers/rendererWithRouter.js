import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import { createMemoryHistory } from "history";
import { StateProvider } from "react-conflux";
import { globalReducer, globalContext } from "../store/reducers/globalReducer";

const rendererWithRouter = (
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...renderer.create(
      <StateProvider reducer={globalReducer} stateContext={globalContext}>
        <Router history={history}>{ui}</Router>
      </StateProvider>
    ),
    history
  };
};

export default rendererWithRouter;
