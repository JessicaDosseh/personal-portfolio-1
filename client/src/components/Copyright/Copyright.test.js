import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { renderWithRouter, rendererWithRouter } from "../../testHelpers";

import Copyright from "./Copyright";

describe("<Copyright />", () => {
  it("should render the Copyright component without crashing", () => {
    const helpers = renderWithRouter(<Copyright darkmode={false} />);
  });

  it("should match the snapshot of Copyright", () => {
    const tree = rendererWithRouter(<Copyright darkmode={false} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it("should render the copyright text to the screen", () => {
    const { getByText } = renderWithRouter(<Copyright darkmode={false} />);
    expect(getByText(/Copyright Â© 2019 Nathan Thomas/i)).toBeDefined();
  });
});
