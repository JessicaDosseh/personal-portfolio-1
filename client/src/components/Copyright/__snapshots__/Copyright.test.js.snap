// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Copyright /> should match the snapshot of Copyright 1`] = `
<div
  className="sc-AykKC bFoCbz"
>
  <h4>
    Copyright Â© 2019 Nathan Thomas
  </h4>
</div>
`;
